solidity/contracts/utility/MocBTCToUSDOracle.sol

```diff
@@ -36,10 +37,12 @@ contract MocBTCToUSDOracle is IConsumerPriceOracle, Owned {
     /**
       * @dev returns the USD/BTC update time.
       *
-      * @return always returns current block's timestamp
+      * @return returns the latest block's timestamp
     */
     function latestTimestamp() external view returns (uint256) {
-        return now; // MoC oracle doesn't return update timestamp
+        require(block.number >= Medianizer(mocOracleAddress).getLastPublicationBlock(), "latest block number larger than current block numer");
+        uint256 latestTimestamp_ = block.timestamp - ( block.number - Medianizer(mocOracleAddress).getLastPublicationBlock() ) * 30;
```

Comment:
The block time (30s) might change in the future. Please make a variable ```blockTime```, which is set to 30s by default but can be set by the owner.